src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs

```diff
@@ -372,5 +444,103 @@ End Class
 
             await VerifyVB.VerifyAnalyzerAsync(source);
         }
+
+        [Theory]
+        [InlineData("Dictionary<ISymbol, string>", "EqualityComparer")]
+        [InlineData("Dictionary<ISymbol, string>", "EqualityComparerAndComparer")]
+        [InlineData("SortedDictionary<ISymbol, string>", "SimpleComparer")]
+        [InlineData("SortedDictionary<ISymbol, string>", "EqualityComparerAndComparer")]
+        [InlineData("SortedSet<ISymbol>", "SimpleComparer")]
+        [InlineData("SortedSet<ISymbol>", "EqualityComparerAndComparer")]
+        [InlineData("HashSet<ISymbol>", "EqualityComparer")]
+        [InlineData("HashSet<ISymbol>", "EqualityComparerAndComparer")]
+        public async Task CompareSymbolWithComparisonTypes_WithComparer_CSharp(string dataType, string comparisonType)
+        {
+            var source = $@"
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+class TestClass {{
+    private {dataType} testData = new {dataType}({comparisonType}.Instance);
+}}
+";
+            await new VerifyCS.Test()
+            {
+                TestState = { Sources = { source, MinimumComparerDeclarationCSharp } }
+            }.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("Dictionary<ISymbol, string>", 52)]
```

Comment:
Is the column here just some fixed constant plus dataType.Length? Might be easier to inline the computation than have magic numbers.